libs/permit-utils/vite.config.ts

```diff
@@ -0,0 +1,52 @@
+/// <reference types="vitest" />
+import { defineConfig } from 'vite'
+import macrosPlugin from 'vite-plugin-babel-macros'
+import dts from 'vite-plugin-dts'
+import viteTsConfigPaths from 'vite-tsconfig-paths'
+
+import * as path from 'path'
+
+export default defineConfig({
+  cacheDir: '../../../node_modules/.vite/permit-utils',
+
+  plugins: [
+    dts({
+      entryRoot: 'src',
+      tsConfigFilePath: path.join(__dirname, 'tsconfig.lib.json'),
+      skipDiagnostics: true,
+    }),
+
+    macrosPlugin(),
+
+    viteTsConfigPaths({
+      root: '../../../',
+    }),
+  ],
+
+  // Uncomment this if you are using workers.
+  // worker: {
+  //  plugins: [
+  //    viteTsConfigPaths({
+  //      root: '../../',
+  //    }),
+  //  ],
+  // },
+
+  // Configuration for building your library.
+  // See: https://vitejs.dev/guide/build.html#library-mode
+  build: {
+    lib: {
+      // Could also be a dictionary or array of multiple entry points.
+      entry: 'src/index.ts',
+      name: 'permit-utils',
+      fileName: 'index',
+      // Change this to the formats you want to support.
+      // Don't forget to update your package.json as well.
+      formats: ['es', 'cjs'],
+    },
+    rollupOptions: {
+      // External packages that should not be bundled into your library.
+      external: [/@1inch/, /@cowprotocol/, /@ethersproject/],
```

Comment:
Explicitly removing third party dependencies to not have them included in the bundle.
They are instead added as part of package.json dependencies.